aiohttp/middlewares.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+import re
+
+from aiohttp.web_exceptions import HTTPMovedPermanently
+from aiohttp.web_urldispatcher import SystemRoute
+
+
+@asyncio.coroutine
+def _check_request_resolves(request, path):
+    alt_request = request.clone(path=path)
+
+    match_info = yield from request.app.router.resolve(alt_request)
+    alt_request._match_info = match_info
+
+    if not isinstance(match_info.route, SystemRoute):
+        return True, alt_request
+
+    return False, request
+
+
+def normalize_path(
```

Comment:
Rename to `normalize_path_middleware`. The name is long but it will be used only in one place of program.
To adding `middleware` suffix explicitly describes the function job.